### PR TITLE
Scope args fix

### PR DESCRIPTION
**Pull request by anderson-joyle**
Fixing a possible order issue when passing scope types to `scopeInfo.CheckTypes`.

### CODE REVIEW

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — DISMISSED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -4469,14 +4469,15 @@ public override bool PreVisit(CallNode node)
                         }
                         else if (carg > 0)
                         {
-                            var types = new DType[maybeFunc.ScopeArgs];
+                            // Gets the lesser number between the CallNode chidl args and func.ScopeArgs.
+                            // There reason why is that the intellisense can visit this code and provide a number of args less than the func.ScopeArgs.
+                            // Example: Join(|
+                            var argsCount = Math.Min(node.Args.Children.Count, maybeFunc.ScopeArgs);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4475]: This fixesd #2759

**Pull request merged**